### ISSUE TITLE
Colon Rule is not triggered with improperly formatted colons in 'rethrows' function signature

### ISSUE THREAD

**Issue by maximkhatskevich** [realm/SwiftLint#2459]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Colon Rule does not trigger with improperly formatted colons in function parameters list (between param name and type) when function is marked as 'rethrows'.

##### Complete output when running SwiftLint, including the stack trace and command used

N/A

### Environment

* SwiftLint version - 0.27.0
* Installation method used - CocoaPods
* Paste your configuration file:

```yml
# see docs at https://github.com/realm/SwiftLint

disabled_rules:
  - function_parameter_count
  - trailing_newline
  - closure_parameter_position
  - opening_brace
  - nesting
  - function_body_length
  - file_length

included: # paths to include during linting. `--path` is ignored if present.
  - Sources
  - Tests

# rules options:

line_length:
  ignores_comments: true
trailing_whitespace:
  ignores_empty_lines: true
type_name:
  allowed_symbols: _
statement_position:
  # https://github.com/realm/SwiftLint/issues/1181#issuecomment-272445593
  statement_mode: uncuddled_else
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?

  NO

* Which Xcode version are you using (check `xcode-select -p`)?

Xcode 10.0

* Do you have a sample that shows the issue?

```swift
// This does NOT trigger a violation:
func someFunc(
        _ input :Int?, // should be a Colon Violation in this line, but nothing happens...
        _ completion: () throws -> Void
        ) rethrows
{
    //
}
```

```swift
// This triggers a violation:
func someFunc(
        _ input :Int?, // Colon Violation: Colons should be next to the identifier... 
        _ completion: () throws -> Void
        )
{
    //
}
```

**Comment by marcelofabri** [comment]
I looked into this and to my surprise, this is not a bug caused by our regex! It looks SourceKit returns the wrong syntax type (`identifier` instead of `typeidentifier`) for types in the function declaration when `rethrows` is present.

I've filled https://bugs.swift.org/browse/SR-9576 to track this on SourceKit itself.

**Comment by marcelofabri** [comment]
https://github.com/apple/swift/pull/22017 was merged, so this should be fixed in a future Swift version (not 5.0 though).

**Comment by marcelofabri** [comment]
This was released as part of Swift 5.1 🚀
